zebra/redistribute.c

```diff
@@ -313,14 +313,18 @@ void redistribute_delete(const struct prefix *p, const struct prefix *src_p,
 
 		/* Send a delete for the 'old' re to any subscribed client. */
 		if (old_re
-		    && (vrf_bitmap_check(client->redist[afi][ZEBRA_ROUTE_ALL],
-					 old_re->vrf_id)
+		    && ((is_default_prefix(p)
+			 && vrf_bitmap_check(client->redist_default[afi],
+					     old_re->vrf_id))
+			|| (vrf_bitmap_check(
```

Comment:
nit: redundant parentheses around `vrf_bitmap_check()`

Reply:
Thanks. Will fix it.